### PR TITLE
Friendships Milestone 3

### PR DESCRIPTION
**Pull request by karvel-code**
In this section I have:
- Updated the Rails routes.
- Ensured that all the features are working correctly

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -18,6 +18,14 @@ def destroy
...
+  #   @friendship = Friendship.find_by user_id: current_user.id, friend_id: params[:id]
+  #   @friendship_reverse = Friendship.find_by friend_id: current_user.id, user_id: params[:id]
+  #   @friendship&.destroy
+  #   @friendship_reverse&.destroy
+  #   redirect_to users_path
+  # end
+
   def confirm
     @friendship = Friendship.find_by(id: params[:id])
     @friendship.update_attribute(:confirmed, true)
```
Comment (Elbie-em) [inline app/controllers/friendships_controller.rb]: - [ ] when friendship is confirmed, an additional row with reversed user-friend pair is created, which means your table should have a reversed row created upon confirmation. e.g
  id:1  user_id: 1 friend_id: 2 ,confirmed: true
  id:2  user_id 2 friend_id 1, confirmed: true
The friendship IDs do not need to be contiguous as users may accept requests at a later time.
The goal is to allow users to access their friends by used the friend_id.

File: app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -18,6 +18,14 @@ def destroy
...
   end
 
+  # def remove_friendship
+  #   @friendship = Friendship.find_by user_id: current_user.id, friend_id: params[:id]
+  #   @friendship_reverse = Friendship.find_by friend_id: current_user.id, user_id: params[:id]
+  #   @friendship&.destroy
+  #   @friendship_reverse&.destroy
+  #   redirect_to users_path
+  # end
+
```
Comment (Elbie-em) [inline app/controllers/friendships_controller.rb]: - [ ] Kindly remove commented code, this will make your controller look more clean and professional.

**Review by Elbie-em — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED 🟡  ♻️ 

Hi @karvel-code, this is a very impressive approach you have taken in creating the inverse friendship functionality. However, the milestone initial requirement has not been met. Please check my comments below ⬇️  and make the necessary changes in order for this milestone to be approved.

### [OPTIONAL]

- [ ] The user's view is showing all users including the current user, you could simply create a query that shows all users except the current user in your controller.
- [ ] The Pending  link can be an action to cancel the request.
- [ ] If the users are friends, the Friends link can be an action to cancel the friendship.

*All the best and good luck* 💪 

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coder, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The app runs just fine. 👍🏽 


### **_Room for Improvement_** 🤪 

- [ ] One of the advantages of using SQL queries instead of ruby enumerables to access the database is the ease and speed. Ruby enumerables are just not built for this purpose. Remove the ruby methods in the user model and stick to using only the SQL queries. You currently have a copy of two database queries( SQL queries and ruby enumerables).

### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**